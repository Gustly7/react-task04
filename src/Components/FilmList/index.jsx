import React, { Component } from 'react'
import Film from '../Film/Film'

var films = [
  {filmName: 'Kill Bill vol. 1', filmYear: '2003', filmGenre: 'Action & Adventure', id: 1},
  {filmName: 'Kill Bill vol. 2', filmYear: '2004', filmGenre: 'Action & Adventure', id: 2},
  {filmName: 'Kill Bill vol. 3', filmYear: '2005', filmGenre: 'Action & Adventure', id: 3},
  {filmName: 'Kill Bill vol. 4', filmYear: '2006', filmGenre: 'Action & Adventure', id: 4}
]

class FilmList extends Component {
  render () {
    return (
      <div className='container filmList'>
        {films.map((film, i) => {
          console.log('Entered')
          // Return the element. Also pass key
          return (<Film key={film.id} filmName={film.filmName} filmYear={film.filmYear} filmGenre={film.filmGenre} />)
        })}
      </div>
    )
  }
}

export default FilmList
