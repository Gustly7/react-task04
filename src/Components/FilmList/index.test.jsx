import React, { Component } from 'react'
import FilmList from './index'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('FilmList component test suite', () => {
  test('FilmList snapshot renders correctly', () => {
    const tree = renderer.create(
      <FilmList />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('FilmList div should have "container filmList" classes', () => {
    const wrapper = shallow(
      <FilmList />
    )
    const elem = wrapper.find('div.container.filmList')
    expect(elem.length).toBe(1)
  })
})
