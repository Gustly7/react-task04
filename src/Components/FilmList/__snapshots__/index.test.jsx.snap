// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList component test suite FilmList snapshot renders correctly 1`] = `
<div
  className="container filmList"
>
  <div
    className="Film"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="Poster"
          >
            <img
              alt={undefined}
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="FilmName"
          >
            Kill Bill vol. 1
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="FilmYear"
          >
            2003
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="FilmGenre"
          >
            Action & Adventure
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Film"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="Poster"
          >
            <img
              alt={undefined}
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="FilmName"
          >
            Kill Bill vol. 2
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="FilmYear"
          >
            2004
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="FilmGenre"
          >
            Action & Adventure
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Film"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="Poster"
          >
            <img
              alt={undefined}
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="FilmName"
          >
            Kill Bill vol. 3
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="FilmYear"
          >
            2005
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="FilmGenre"
          >
            Action & Adventure
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Film"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="Poster"
          >
            <img
              alt={undefined}
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="FilmName"
          >
            Kill Bill vol. 4
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="FilmYear"
          >
            2006
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="FilmGenre"
          >
            Action & Adventure
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
